File: 443-string-compression/string-compression.java
class Solution {
    public int compress(char[] chars) {
        int n = chars.length;
        int count = 0;
        int index = 0;

        for(int i = 0;i<n;i++){
                    count++;
                   if(i+1 == n || chars[i] != chars[i+1]){
                    chars[index++] = chars[i];
                   if(count != 1){
                    for(char c:Integer.toString(count).toCharArray()){
                        chars[index++] = c;
                    }
                   }
                                      count = 0;
                   }
                }
return index;
                }
            }

File: 443-string-compression/string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        n = len(chars)
        count = 0
        index = 0
        for i in range(n):
            count += 1
            if i + 1 == n or chars[i] != chars[i + 1]:
                chars[index] = chars[i]
                index += 1
                if count > 1:
                    for ch in str(count):
                        chars[index] = ch
                        index += 1
                count = 0
        return index
